app/helpers/puppetca_helper.rb

```diff
@@ -1,8 +1,19 @@
 module PuppetcaHelper
+  STATES = [N_('revoked'), N_('pending'), N_('valid')]
+
   def state_filter
-    states = [N_('revoked'), N_('pending'), N_('valid')]
-    select_tag "Filter", options_for_select([''] + states.map { |s| [_(s), s] }, params[:state]), :class => "puppetca-filters",
-               :onchange => "window.location.href = '#{smart_proxy_puppetca_index_path(@proxy)}' + (this.value == '' ? '' : ('?state=' + this.value))"
+    select_tag "Filter", options_for_select([[_('valid or pending'), _('valid')+'|'+_('pending')]] +
```

Comment:
since the filtering is done by datatables on the table data itself, the states will already be translated, so we need to have the filter use the translated strings as well, and not the original strings as was before.